ibm/resource_ibm_satellite_cluster.go

```diff
@@ -0,0 +1,766 @@
+	d.Set("private_service_endpoint_url", *cluster.ServiceEndpoints.PrivateServiceEndpointURL)
+	d.Set("private_service_endpoint_enabled", *cluster.ServiceEndpoints.PrivateServiceEndpointEnabled)
+	d.Set("public_service_endpoint_enabled", *cluster.ServiceEndpoints.PublicServiceEndpointEnabled)
+
+	var zones = make([]map[string]interface{}, 0)
+	for _, zone := range cluster.WorkerZones {
+		zoneInfo := map[string]interface{}{
+			"id": zone,
+		}
+		zones = append(zones, zoneInfo)
+	}
+	d.Set("zones", zones)
+
+	tags, err := GetTagsUsingCRN(meta, *cluster.Crn)
+	if err != nil {
+		log.Printf(
+			"An error occured during reading of instance (%s) tags : %s", d.Id(), err)
+	}
+	d.Set("tags", tags)
+
+	return nil
+}
+
+func resourceIBMSatelliteClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	clusterID := d.Id()
+	workerPoolName := "default"
+
+	satClient, err := meta.(ClientSession).SatelliteClientSession()
+	if err != nil {
+		return err
+	}
+
+	csClient, err := meta.(ClientSession).ContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	targetEnv, err := getClusterTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	var resourceGroup string
+	if v, ok := d.GetOk("resource_group_id"); ok {
+		resourceGroup = v.(string)
+	}
+
+	wrkAPI := csClient.Workers()
+	clusterAPI := csClient.Clusters()
+	if (d.HasChange("kube_version") || d.HasChange("patch_version") || d.HasChange("retry_patch_version")) && !d.IsNewResource() {
+		if d.HasChange("kube_version") {
+			var masterVersion string
+			if v, ok := d.GetOk("kube_version"); ok {
+				masterVersion = v.(string)
+			}
+			params := v1.ClusterUpdateParam{
+				Action:  "update",
+				Force:   true,
+				Version: masterVersion,
+			}
+			err := clusterAPI.Update(clusterID, params, targetEnv)
+			if err != nil {
+				return err
+			}
+			_, err = WaitForSatelliteClusterVersionUpdate(d, meta, targetEnv)
+			if err != nil {
+				return fmt.Errorf(
+					"Error waiting for cluster (%s) version to be updated: %s", d.Id(), err)
+			}
+		}
+		satClient, err := meta.(ClientSession).SatelliteClientSession()
+		if err != nil {
+			return err
+		}
+		workerOpts := satClient.NewGetWorkers1Options(clusterID)
+		workerFields, response, err := satClient.GetWorkers1(workerOpts)
+		if err != nil {
+			return fmt.Errorf("Error retrieving workerFields %s: %s", err, response)
+		}
+
+		getSatClusterOptions := &kubernetesserviceapiv1.GetClusterOptions{
+			Cluster:            &clusterID,
+			XAuthResourceGroup: &resourceGroup,
+		}
+
+		cluster, response, err := satClient.GetCluster(getSatClusterOptions)
+		if err != nil {
+			return fmt.Errorf("Error retrieving cluster %s: %s", clusterID, err)
+		}
+		waitForWorkerUpdate := d.Get("wait_for_worker_update").(bool)
+
+		for _, w := range workerFields {
+			//kubeversion update done if there is a change in Major.Minor version
+			if strings.Split(*w.KubeVersion.Actual, "_")[0] != strings.Split(*cluster.MasterKubeVersion, "_")[0] {
```

Comment:
I think we need to add more additional condition similar to container cluster

Reply:
Added additional parameter to validate